nats/aio/client.py

```diff
@@ -120,6 +120,26 @@ def messages(self):
 
         return self._message_iterator
 
+    async def next_msg(self, timeout=0.5):
+        """
+        next_msg can be used to retrieve the next message
+        from a stream of messages using await syntax.
+        """
+        future = asyncio.Future()
+
+        async def _next_msg():
+            msg = await self._pending_queue.get()
+            future.set_result(msg)
+            
+        task = asyncio.create_task(_next_msg())
+        try:
+            msg = await asyncio.wait_for(future, timeout)
```

Comment:
For the implementation was looking at maybe using an async generator but for since there is a timeout involved went with getting directly from the pending queue instead for now.